### PR TITLE
Updated react-maskedinput for 4.0.0

### PR DESCRIPTION
**Pull request by LavaToaster**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/insin/react-maskedinput/commit/09472b84e83fe7f0d2ea693125731a306a7b3177
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

3.3.4 => 4.0.0 changes it so that it does have a default export, other than that there were no breaking changes to anything else.

### CODE REVIEW

**Comment by Lavoaster** [comment]
This module has two differing exports based on what is being loaded. This PR resolves an issue loading it with the new `es` module it has, but conflicts with the `lib` module. You can read more about the issue in insin/react-maskedinput#104.

I'm not sure what to do about this PR, is this something I should / could resolve here? Or would it be better to wait for the package maintainer to make a decision on if exports should be standard or not?

File: types/react-maskedinput/index.d.ts
Comment (LKay) [inline types/react-maskedinput/index.d.ts]: Don't rename interfaces when you only changed the way they are exported. This will break people's code.

File: types/react-maskedinput/index.d.ts
Comment (LKay) [inline types/react-maskedinput/index.d.ts]: You can just do:
```typescript
declare class MaskedInput extends React.Component<MaskedInputProps> {}
export default MaskedInput;
```

**Review by LKay — CHANGES_REQUESTED** [review]

File: types/react-maskedinput/index.d.ts
Comment (Lavoaster) [inline types/react-maskedinput/index.d.ts]: I assumed this would be fine with the potential breaking change introduced in 4.0.0. Will revert though.

**Comment by Lavoaster** [comment]
@LKay I've made the changes you've request.

Do you have any thoughts on my previous comment?

**Comment by LKay** [comment]
I think you misunderstood me. It's ok to remove namespace. Since the export in not in cjs style you can export all the types and interfaces directly. You should avoid using `export as namespace ...`. Just add `export` before interfacesAfter you remove namespace.

**Comment by RyanCavanaugh** [comment]
@LKay - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Comment by Lavoaster** [comment]
@LKay Oh! I get it now. Sorry, about that.

Just FYI, the package exports both CJS and ES modules, but they don't interop with this definition.

**Comment by LKay** [comment]
Now it's LGTM and ready to merge.

**Review by LKay — APPROVED** [review]

**Pull request merged**
